hswidget/ds1widget/state.go

```diff
@@ -0,0 +1,108 @@
+package ds1widget
+
+import (
+	"github.com/OpenDiablo2/HellSpawner/hswidget"
+)
+
+type ds1EditorMode int
+
+const (
+	ds1EditorModeViewer ds1EditorMode = iota
+	ds1EditorModeAddFile
+	ds1EditorModeAddObject
+	ds1EditorModeAddPath
+	ds1EditorModeAddFloorShadow
+	ds1EditorModeAddWall
+	ds1EditorModeConfirm
+)
+
+type ds1Controls struct {
+	tileX, tileY int32
+	object       int32
+	// nolint:unused,structcheck // will be used
+	subgroup int32
+	// nolint:unused,structcheck // will be used
+	tile struct {
+		floor, wall, shadow, sub int32
+	}
+}
+
+// DS1AddObjectState represents state of new object
+type DS1AddObjectState struct {
+	objType  int32
+	objID    int32
+	objX     int32
+	objY     int32
+	objFlags int32
+}
```

Comment:
is there any reason for these types to be exported?

Reply:
any one, forgot to change